### DISCUSSION TITLE
Welcome to CODES Discussions!

### DISCUSSION THREAD

**Discussion by nmcglo** [codes-org/codes#213] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

This will be a place for broad discussions and help to be given out. If you believe you've encountered a bug, open an issue! But if you just have a question, feel free to start a discussion here!

**Comment by sappyb** [comment]
Hi, I have a DUMPI trace which has BARRIER. How can I replay the DUMPI trace with the barrier operation? I am calling the model-net-mpi-replay for replaying my traces.

**Reply by nmcglo** [reply]
Unfortunately, CODES trace replaying does not currently support MPI_BARRIER and I no longer develop for CODES. To enable support for it, you'd have to develop the necessary features in `src/workload/methods/codes-dumpi-trace-nw-wrkld.c` and `src/network-workloads/model-net-mpi-replay.c`. 

Also, the development of the supported DUMPI/MPI trace operations well predates my onboarding to the project in 2016 so I don't have a good idea on how exactly this would be implemented (apart from where said trace operation handling is done as mentioned above) and as of Oct. 2022, I took a new job and no longer actively develop for CODES.

**Comment by sahityagorthi** [comment]
Hey !! I want to run the DUMPI traces, but when I run , I come across the error : failed to find workload generator dumpi-trace-workload, when I dont provide any allocation and workload conf file in the command.
When the allocation and workload conf files are mentioned in the cmmand then, I get the following o/p on the terminal:
ROSS Version: v8.0.0



tw_net_start: Found world size to be 1 

node: 0: error: ../src/network-workloads/model-net-mpi-replay.c:3339: 

 Could not open file ../src/network-workloads/workloads.conf 

--------------------------------------------------------------------------

MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD

with errorcode 1.



NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.

You may or may not see output from other processes, depending on

exactly when Open MPI kills them.

Please tell me how to resolve the issues

**Reply by nmcglo** [reply]
You should be able to supply just the base path to the dumpi trace you're trying to run with the following arguments

For an example dumpi trace with 216 ranks
```
--num_net_traces=216 --workload_file=/path/to/dumpi/trace/directory/dumpi-2014.03.03.15.09.03-
--workload_type="dumpi"
```

You'll still need the modelnet config file that configures the network.

For more complex workloads you'll need to create the above mentioned workload configuration file.

This file looks like:
`example-workload.conf:`
```
100 synthetic1
125 /path/to/dumpi/trace/directory/Multigrid/MultiGrid_C_n125_dumpi/dumpi-2014.03.06.23.48.13-
```

This for instance runs two workloads on the network, one synthetic (uniform random) with 100 ranks, and one dumpi workload (multigrid 125 as an example) with 125 ranks.

To do this you will also need an allocation file dictating the placement of each rank of the workload. Instructions for generating that file can be found here: https://github.com/codes-org/codes/wiki/codes-allocations

you supply that allocation conf file as `--alloc-file=/path/to/generated/file.alloc`
